src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -522,50 +524,69 @@ public SCM build(SCMHead head, SCMRevision revision) {
                 }
             }
         }
+        String browserUrl = buildBitbucketClient().getRepositoryUri(repositoryType, BitbucketRepositoryProtocol.HTTP, null, repoOwner, repository);
         if (head instanceof PullRequestSCMHead) {
             PullRequestSCMHead h = (PullRequestSCMHead) head;
             if (repositoryType == BitbucketRepositoryType.MERCURIAL) {
                 MercurialSCM scm = new MercurialSCM(getRemote(h.getRepoOwner(), h.getRepository(),
                         BitbucketRepositoryType.MERCURIAL));
                 // If no revision specified the branch name will be used as revision
-                scm.setRevision(revision instanceof MercurialRevision
-                        ? ((MercurialRevision) revision).getHash()
-                        : h.getBranchName()
-                );
-                scm.setRevisionType(RevisionType.BRANCH);
+                if (revision instanceof MercurialRevision) {
```

Comment:
@jglick This is how I fixed the changelog not generating, I would like to go through my findings with you and see if there are no unforeseen consequences.

The oly one I can see is that the first build for any user after an upgrade will not get a changelog, but the next one should generate one. OTOH for users that has been running the refactored version before won't have a changelog anyway :)